polaris-react/src/components/LegacyFilters/components/ConnectedFilterControl/ConnectedFilterControl.scss

```diff
@@ -110,6 +116,8 @@
 }
 
 .MoreFiltersButtonContainer.onlyButtonVisible {
+  padding-left: 0;
```

Comment:
@lgriffee I think this should [address your feedback](https://github.com/Shopify/polaris/pull/11294#issuecomment-1846256853) and make sure that these changes don't affect the spacing at larger breakpoints